torch_geometric/explain/algorithm/gnn_explainer.py

```diff
@@ -184,40 +335,135 @@ def _initialize_masks(self, x: Tensor, edge_index: Tensor):
         else:
             assert False
 
+    def _collect_gradients(self):
+        if self.is_hetero:
+            # Collect gradients for heterogeneous graph
+            self._collect_hetero_gradients()
+        else:
+            # Collect gradients for homogeneous graph
+            self._collect_homo_gradients()
```

Comment:
I'm in favor of not adding redundant comments that don't add any info:
```suggestion
        if self.is_hetero:
            self._collect_hetero_gradients()
        else:
            self._collect_homo_gradients()
```